pkg/component/controller/cplb_unix.go

```diff
@@ -71,20 +78,45 @@ func (k *Keepalived) Init(_ context.Context) error {
 
 // Start generates the keepalived configuration and starts the keepalived process
 func (k *Keepalived) Start(_ context.Context) error {
-	if k.Config == nil {
+	if k.Config == nil || (len(k.Config.VRRPInstances) == 0 && len(k.Config.VirtualServers) == 0) {
+		k.log.Warn("No VRRP instances or virtual servers defined, skipping keepalived start")
 		return nil
 	}
 
-	if err := k.configureDummy(); err != nil {
-		return fmt.Errorf("failed to configure dummy interface: %w", err)
+	if len(k.Config.VRRPInstances) > 0 {
+		if err := k.configureDummy(); err != nil {
+			return fmt.Errorf("failed to configure dummy interface: %w", err)
+		}
+		if err := k.Config.ValidateVRRPInstances(nil); err != nil {
```

Comment:
Again, IMO we need to hook validation to general config validation "phase"

Reply:
Agreed, it's added as a separate commit because rewriting the git history wasn't easy...